projects/project.cpp

```diff
@@ -208,11 +213,11 @@ namespace projects {
       
       for (size_t p=0; p<getObjectWrapper().particleSpecies.size(); ++p) {
          this->setVelocitySpace(p,cell);
+         uint n = cell->get_number_of_velocity_blocks(p);
+         if ( n < 64) {
+            cout << " WARNING!!! Low block count in this cell for population " << getObjectWrapper().particleSpecies[p].name << ". Number of velocity blocks = " << n << endl;
+         }
       }
-
-      //let's get rid of blocks not fulfilling the criteria here to save memory.
-      //cell->adjustSingleCellVelocityBlocks();
```

Comment:
Now this is interesting. Apparently at some point we found it useful not to adjust here any more? Why?

Reply:
https://github.com/fmihpc/vlasiator/blob/d8c3a01b62aa40c2914fe1b9140acbde0b4bd38e/spatial_cell.cpp#L368
It still exists, only takes popID nowadays.

Reply:
That function is used to drop blocks that are below the threshold. A wrapper doing it for all cells will be called after initialisation has completed, I'm pretty sure, but not calling it before doing the moments surprises me a bit.